eng/scripts/Update-PackageVersion.ps1

```diff
@@ -66,6 +66,40 @@ if ($packageSemVer.HasValidPrereleaseLabel() -ne $true) {
   exit 1
 }
 
+function ReplacePathDependencyVersions() {
+  $cargoTomlFiles = Get-ChildItem -Path . -Recurse -Filter Cargo.toml
+  | Where-Object FullName -NotMatch "$RepoRoot/(target|eng)/*"
+
+  foreach ($cargoTomlFile in $cargoTomlFiles) {
+    $content = Get-Content -Path $cargoTomlFile.FullName -Raw
+    $updated = $content
+
+    if ($content -match "\n\s*\[(.+[\.-])?dependencies\.$PackageName\][^\[]+") {
+      # if there is a dependency block containing a path and version for the package, update the version
+      $dependencyBlock = $matches[0]
+      if ($dependencyBlock -match '\n\s*path\s*=') {
+        $replacement = $dependencyBlock -replace '(\n\s*version\s*=\s*)".*"', "`$1`"$packageSemVer`""
+        $updated = $updated.Replace($dependencyBlock, $replacement)
```

Comment:
Don't Repeat Yourself (DRY): you could return either as the line e.g.,

```powershell
$dependency = if ($content -match ... -and $matches[0] -match '\n...') {
  $matches[0]
elseif ($content -match ... -and $matches[0] -match '[...') {
  $matches[0]
}
if ($dependency) {
  $replacement = ...
}
```

Reply:
Updated to use rust and toml_edit,